refactor: use method signature for class overloads (#377)
remirror/remirror: .changeset/cool-countries-pay.md

---

> '@remirror/core': minor

Author: Would be awesome if you could also add every extension that is affected by the changes.

'@remirror/extension-bidi': patch
'@remirror/extension-placeholder': patch
---

> Remove the first parameter `extensions` from method `onCreate`, `onView` and `onDestroy`.

Author: ```suggestion
Remove the first parameter `extensions` from the lifecycle methods `onCreate`, `onView` and `onDestroy`.
```

> Switch to using method signatures for extension class methods as discussed in #360. The follow methods have been affected:

Author: ```suggestion
Switch to using method signatures for extension class methods as discussed in #360. The following methods have been affected:
```

---
'@remirror/core': minor
'@remirror/extension-bidi': patch
'@remirror/extension-placeholder': patch
---

Switch to using method signatures for extension class methods as discussed in #360.

> Remove the first parameter `extenisons` from method `onCreate`, `onView` and `onDestroy`.

Author: ```suggestion
Remove the first parameter `extensions` from method `onCreate`, `onView` and `onDestroy`.
```
